### PR TITLE
Arrays

### PR DESCRIPTION
**Pull request by RabeaGleissner**
Hi @rouanw 
I think array support would be very useful for the project I'm currently working on! 😄 
What do you think about this suggestion?

### CODE REVIEW

File: randoc.js [hunk 1]
```diff
@@ -2,7 +2,16 @@ const chance = new require('chance')();
...
+    return {
+      ...(schema._arrayOf && { [fieldName]: [] })
+    };
+  }
+  if (schema._arrayOf !== undefined && chance[schema._type]) {
+    const returnArray = []
+    for (let i = 1; i <= schema._arrayOf; i++) {
+      returnArray.push(chance[schema._type](schema.args))
+    }
+    return { [fieldName]: returnArray };
```
Comment (rouanw) [inline randoc.js]: What about something like:

```js
if (schema._arrayOf !== undefined && chance[schema._type]) {
    return { [fieldName]: [...new Array(schema._arrayOf)].map(_ => chance[schema._type](schema.args)) };
}
```

File: randoc.js [hunk 2]
```diff
@@ -2,7 +2,16 @@ const chance = new require('chance')();
 
 const specialRandomDocument = (fieldName, schema) => {
   if (schema._exists && !chance.bool({ likelihood: schema._exists }) ) {
-    return {};
+    return {
+      ...(schema._arrayOf && { [fieldName]: [] })
+    };
+  }
+  if (schema._arrayOf !== undefined && chance[schema._type]) {
```
Comment (rouanw) [inline randoc.js]: Elsewhere this library [defaults to `string`](https://github.com/rouanw/randoc/blob/d9c3cdb9c9846f3bd5a2b7cc9a98bc64ddb4b58c/randoc.js#L28-L29) if the type doesn't end up being a real chance.js function. Maybe we could extract that logic and use it wherever we call chance with a type that may or may not exist?

**Review by rouanw — APPROVED** [review]
Thanks @RabeaGleissner! Looks good.

File: randoc.js (hunk 1 above)
Comment (RabeaGleissner) [reply randoc.js]: Nice! I'll update it 👍

File: randoc.js (hunk 2 above)
Comment (RabeaGleissner) [reply randoc.js]: Sounds good. I made that change as well. But for the array I'm still returning an array with one Change string in it, so it's the same type.

**Pull request merged**